gluten-core/src/main/scala/io/glutenproject/expression/TernaryExpressionTransformer.scala

```diff
@@ -108,6 +108,39 @@ case class StringLocateTransformer(
   }
 }
 
+case class StringSplitTransformer(
+    substraitExprName: String,
+    srcExpr: ExpressionTransformer,
+    regexExpr: ExpressionTransformer,
+    limitExpr: ExpressionTransformer,
+    original: StringSplit)
+  extends ExpressionTransformer
+  with Logging {
+
+  override def doTransform(args: java.lang.Object): ExpressionNode = {
+    // In velox, split function just support tow args, not support limit arg for now
+    if (!BackendsApiManager.isVeloxBackend) {
+      return TernaryExpressionTransformer(
+        substraitExprName,
+        srcExpr,
+        regexExpr,
+        limitExpr,
+        original).doTransform(args)
+    }
+
+    val limit = limitExpr.doTransform(args).asInstanceOf[IntLiteralNode].getValue
```

Comment:
If `limitExpr` is not literal, an unexpected exception will be thrown, right? For velox, it may need to fall back for non-literal cases.
Ditto for `regexExpr`.

Reply:
Done